xarray/core/dataset.py

```diff
@@ -270,18 +280,36 @@ def _maybe_chunk(
     name_prefix="xarray-",
     overwrite_encoded_chunks=False,
     inline_array=False,
+    chunked_array_type: str
+    | ChunkManagerEntryPoint = None,  # noqa: F821  # type: ignore[name-defined]
```

Comment:
@headtr1ck do you think you could help me with the mypy failures here? 

All I'm trying to do is type hint this argument using the ABC I define in `parallelcompat.py`, but for some reason neither mypy nor ruff is happy about it. I also can't seem to get them both to completely ignore the corresponding errors either :confused:

Reply:
I think I was being an idiot and not noticing the difference between `ChunkManagerEntryPoint` and `ChunkManagerEntrypoint` :sob: The errors I get now make more sense.